File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy = 0;
        int sell = 1;
        int maxP = 0;

        while(sell < prices.length) {
            if(prices[sell] < prices[buy]) {
                buy = sell;
            }
            else {
                maxP = Math.max(maxP, prices[sell] - prices[buy]);
            }
            sell++;
        }
        return maxP;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self,prices):
        left = 0
        right = 1
        maxP = 0

        while right < len(prices):
            currentP = prices[right] - prices[left]
            if prices[left] < prices[right]:
                maxP = max(maxP, currentP)
            else:
                left = right
            right += 1

        return maxP
            